### ISSUE TITLE
upsample_nearest1d backward not supported error

### ISSUE THREAD

**Issue by bpatter5** [huggingface/candle#1548]
I'm attempting to build a VAE and given that backprop is not yet supported for ConvTranspose1d I attempted to use upsampling paired with a convolution layer. However, when I enter the training loop I continue to get the following error:

Error: backward is not supported for upsample-nearest1d.

I'm using candle v 0.3.2 from crates.io and I am attempting to run the training loop on CPU. Happy to provide the stack trace as well if that's helpful. Any insights would be greatly appreciated.

**Comment by bpatter5** [comment]
Closing issue as I should be able to reshape my inputs to take advantage of backprop support for upsampling with nearest 2d.

**Comment by Narsil** [comment]
That was on CPU right ?

**Comment by bpatter5** [comment]
Yes, it was.
